build/build.go

```diff
@@ -604,13 +604,6 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 		}
 	}
 
-	// Propagate SOURCE_DATE_EPOCH from the client env
-	if v := os.Getenv("SOURCE_DATE_EPOCH"); v != "" {
-		if _, ok := so.FrontendAttrs["build-arg:SOURCE_DATE_EPOCH"]; !ok {
-			so.FrontendAttrs["build-arg:SOURCE_DATE_EPOCH"] = v
-		}
-	}
```

Comment:
For the case where buildx is used as lib (compose), it is not going to work anymore I guess?

Reply:
Mm you're right. I'm not quite sure if that's a bug or a feature though :thinking: 

I'm leaning towards it's probably best that when used as a library buildx should avoid reading from the environment as much as possible. We also have some minor precedent here - `DOCKER_DEFAULT_PLATFROM`.

Though I wonder about cases like `BUILDX_NO_DEFAULT_ATTESTATIONS`. On the one hand, it makes sense to propagate them, on the other, it could be confusing to the library consumer if the environment overrides the options that they have passed to Buildx.

IMO, in the future we should push all environment variables that affect the build solve request (so not including buildx driver env vars, etc) into the command layer, and then we should provide a utility method to parse environment variables and modify the build options from them:

https://github.com/docker/buildx/blob/4a73abfd645457ee8819855cb1a337ab23f833dc/controller/pb/controller.proto#L47-L73

Does that seem somewhat reasonable? If so, I'll add a TODO in the code - it's tricky to handle right now, since I think with the controller code, I'd expect we'd try and get library consumers of Buildx to use that API instead of using `build.Build`?

Reply:
SGTM :+1:

Reply:
./cc FYI @glours @ndeloof (for compose)